src/main/java/com/systelab/seed/infrastructure/security/RequestRateLimiterFilter.java

```diff
@@ -0,0 +1,105 @@
+package com.systelab.seed.infrastructure.security;
+
+import com.systelab.seed.infrastructure.properties.ApplicationProperties;
+import io.github.resilience4j.ratelimiter.RateLimiterConfig;
+import io.github.resilience4j.ratelimiter.internal.AtomicRateLimiter;
+import java.io.IOException;
+import java.net.URI;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+
+@Provider
+/**
+ * Filter to limit the API rate limiting at application level.
+ * Filters all requests to REST API. In case we want to filter only the secured calls,
+ * we should add the annotation @AuthenticationTokenNeeded at the class level.
+ */
+public class RequestRateLimiterFilter implements ContainerRequestFilter {
+
+
+  @Context
+  private HttpServletRequest httpServletRequest;
+
+  private Logger logger;
+
+  @Inject
+  public void setLogger(Logger logger) {
+    this.logger = logger;
+  }
+
+  private static final long MINUTES_DURATION = ApplicationProperties.getInstance()
```

Comment:
Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.